baseplate/core.py

```diff
@@ -599,6 +598,57 @@ def _t_request(self):  # pylint: disable=method-hidden
         return _t_request
 
 
+class RequestContext:
+    def __init__(self, factories, wrapped=None, span=None):
+        if not wrapped:
+            wrapped = SimpleNamespace()
+
+        super().__getattribute__("__dict__").update(
+            {"trace": span, "_factories": factories, "_wrapped": wrapped, "_cache": {}}
+        )
+
+    def __getattribute__(self, name):
+        __dict__ = super().__getattribute__("__dict__")
+
+        # if we've already instantiated the object, return it immediately
+        cache = __dict__["_cache"]
+        cached_value = cache.get(name)
+        if cached_value:
+            return cached_value
```

Comment:
I think I remember you doing this before for speed:
```
try:
  return cache['name']
except KeyError:
  pass
```

I'm curious what made you go back to using `.get`

Reply:
Turns out in this case at least this is faster. I benchmarked them against eachother and looked at the bytecode involved.

Reply:
Minor (but noticeable) difference. I don't have the numbers now though.

Reply:
I'm guessing it didn't end up actually being faster?